software/base/src/test/java/org/apache/brooklyn/entity/software/base/SoftwareProcessEntityLatchTest.java

```diff
@@ -116,14 +118,36 @@ public void testLaunchLatchBlocks() throws Exception {
         runTestLatchBlocks(SoftwareProcess.LAUNCH_LATCH, ImmutableList.of("setup", "copyInstallResources", "install", "customize", "copyRuntimeResources"));
     }
 
+    @Test
+    public void testStopLatchBlocks() throws Exception {
+        final AttributeSensor<Boolean> stopper = Sensors.newBooleanSensor("stop.now");
+        final BasicEntity triggerEntity = app.createAndManageChild(EntitySpec.create(BasicEntity.class));
+        final MyService entity = app.createAndManageChild(EntitySpec.create(MyService.class)
+                .configure(SoftwareProcess.STOP_LATCH, DependentConfiguration.attributeWhenReady(app, stopper)));
+        
+        final Task<Void> startTask = Entities.invokeEffector(app, app, MyService.START, ImmutableMap.of("locations", ImmutableList.of(loc)));
+        ((EntityLocal)triggerEntity).sensors().set(Attributes.SERVICE_UP, true);
+        startTask.get(Duration.THIRTY_SECONDS);
+
+        final Task<Void> stopTask = Entities.invokeEffector(app, app, MyService.STOP);
+
+        assertEffectorBlockingDetailsEventually(entity, MyService.STOP.getName(), "Waiting for config " + SoftwareProcess.STOP_LATCH.getName());
+
+        app.sensors().set(stopper, true);
+        stopTask.get(Duration.THIRTY_SECONDS);
```

Comment:
Very minor: I have a personal preference to use `Asserts.DEFAULT_LONG_TIMEOUT` rather than hard-coding 30 seconds here (it comes to the same thing, because that default timeout is 30 seconds).